packages/calcite-components/src/components/panel/panel.tsx

```diff
@@ -560,13 +574,25 @@ export class Panel
   }
 
   renderFooterNode(): VNode {
-    const { hasFooterContent, hasFooterActions } = this;
+    const { hasFooterEndContent, hasFooterStartContent, hasFooterContent, hasFooterActions } = this;
 
-    const showFooter = hasFooterContent || hasFooterActions;
+    const showFooter =
+      hasFooterStartContent || hasFooterEndContent || hasFooterContent || hasFooterActions;
 
     return (
       <footer class={CSS.footer} hidden={!showFooter}>
-        <slot key="footer-slot" name={SLOTS.footer} onSlotchange={this.handleFooterSlotChange} />
+        <slot key="footer-slot" name={SLOTS.footer} onSlotchange={this.handleFooterSlotChange}>
```

Comment:
There is no conditional rendering within the `footer` element, so `key` is not needed on any child (see https://stenciljs.com/docs/templating-jsx#conditionals).